examples/v1alpha2/traffic-splitting/traffic-split-1.yaml

```diff
@@ -0,0 +1,18 @@
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: HTTPRoute
+metadata:
+  name: foo-route
+  labels:
+    gateway: prod-web-gw
+spec:
+  hostnames:
+  - foo.example.com
+  rules:
+  - backendRefs:
+    - name: foo-v1
+      port: 8080
+    - name: foo-v2
+      port: 8080
+  - matches:
+    - path:
+        value: test
```

Comment:
I think there's a bit of nuance missing from the v1alpha1 example here. As I recall, a key part of this example was to have 2 different rule, one that matched everything, and one that was included a header match (https://gateway-api.sigs.k8s.io/v1alpha1/guides/traffic-splitting/#canary-traffic-rollout).

To accomplish that, I think you'd need something like this:

```yaml
  rules:
  - backendRefs:
    - name: foo-v1
      port: 8080
  - matches:
    - headers:
      - name: traffic
        value: test
    backendRefs:
    - name: foo-v2
      port: 8080  
```

Reply:
+1 on this one, this is YAML indenting making things confusing yet again.

Reply:
One way to make it more self-documenting would be to write the default match criteria:


```yaml
rules:
  - matches:
    - path:
        type: Prefix
        value: /
    backendRefs:
    - name: foo-v1
      port: 8080
  - matches:
    - path:
        type: Prefix
        value: /
      headers:
      - name: traffic
        value: test
    backendRefs:
    - name: foo-v2
      port: 8080  
```

Reply:
> I think there's a bit of nuance missing from the v1alpha1 example here. As I recall, a key part of this example was to have 2 different rule, one that matched everything, and one that was included a header match (https://gateway-api.sigs.k8s.io/v1alpha1/guides/traffic-splitting/#canary-traffic-rollout).
> 
> To accomplish that, I think you'd need something like this:
> 
> ```yaml
>   rules:
>   - backendRefs:
>     - name: foo-v1
>       port: 8080
>   - matches:
>     - headers:
>       - name: traffic
>         value: test
>     backendRefs:
>     - name: foo-v2
>       port: 8080  
> ```

🙇‍♀️